articles/getting-started-with-dialogs-in-android-kotlin/index.md

```diff
@@ -0,0 +1,380 @@
+
+```java
+override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        // add the following three lines
+        activityMainBinding?.btnShowDefaultDialog?.setOnClickListener {
+            showDefaultDialog()
+        }
+    }
+```
+
+Ensure that there are no errors in your code before running the app. When you click the button, you should get an output similar to the one below.
+
+![empty dialog](/engineering-education/getting-started-with-dialogs-in-android-kotlin/empty-dialog.png)
+
+##### Adding features to a Dialog
+Now that we've created a Dialog, we can add functionalities such as;
+
+- `Icon` - This is a small-sized image that appears alongside the title. It usually makes the dialog more informative and attractive.
+- `Message` - This is the main content of the alert dialog. It gives a short description of the reason for its pop-up or, prompts the user to make a decision before proceeding.
+- `Buttons` - They appear at the bottom part of the dialog. There are three types of buttons namely, `Positive`, `Negative` and `Neutral` buttons. As their names suggest, they are meant to perform the respective actions once clicked. By default, the dialog dismisses once any of the buttons is clicked.
+- `View` - this is a layout used in customizing dialogs. We'll learn about this in a moment.
+
+Meanwhile, let us create a Dialog that implements the above features. Create a vector drawable (icon) named `ic_hello`.
+
+Update `showDefaultDialog()` function to look like the one below.
+
+```java
+private fun showDefaultDialog() {
+        val alertDialog = AlertDialog.Builder(this)
+
+        alertDialog.apply {
+            setIcon(R.drawable.ic_hello)
+            setTitle("Hello")
+            setMessage("I just wanted to greet you. I hope you are doing great!")
+            setPositiveButton("Positive") { _, _ ->
+                toast("clicked positive button")
+            }
+            setNegativeButton("Negative") { _, _ ->
+                toast("clicked negative button")
+            }
+            setNeutralButton("Neutral") { _, _ ->
+                toast("clicked neutral button")
+            }
+        }.create().show()
+    }
+```
+Also, copy and paste the code below just above or below the `showDefaultDialog()` function.
+
+```java
+    private fun toast(text: String) = Toast.makeText(this, text, Toast.LENGTH_SHORT).show()
+```
+
+**Explanation**
+
+The syntax for creating a button requires us to pass in a listener of type `DialogInterface.OnClickListener` in the lambda function. A `DialogInterface` defines a dialog-type class that can be shown, canceled or dismissed, and may have buttons that can be clicked. `Kotlin` simplifies this by allowing us to pass in `underscores` as arguments of the lambda parameters.
```

Comment:
We use underscores when the parameter is not going to be used in the lambda function. Please incorporate that.